_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,650 @@
+
+Lucene also provides the capability to operate its `KnnVectorQuery` over a subset of documents. To learn more about Lucene’s new capability, see the [Apache Lucene Documentation](https://issues.apache.org/jira/browse/LUCENE-10382).
+
+To learn more about all available k-NN search approaches, including approximate k-NN, exact k-NN with script score and pre-filtering with painless extensions, see [k-NN]({{site.url}}{{site.baseurl}}/opensearch/search-plugins/knn/index/).
+
+### Filtered search performance
+
+Filtering that is tightly integrated with the Lucene HNSW algorithm implementation allows you to to apply k-NN searches more efficiently, both in terms of relevancy of search results and performance. Consider, for example, an exact search using post-filtering on a large data set that returns results slowly and does not guarantee the required number of results specified by `k`.
+With this new capability, you can create an approximate k-NN search, apply filters, and get the number of results that you need. To learn more about approximate searches, see [Approximate k-nn search]({{site.url}}{{site.baseurl}}/opensearch/search-plugins/knn/approximate-knn/).
+
+The HSNW algorithm decides which type of filtering to apply to a search based on the volume of documents, and number of `k` points in the index that you search with a filter.
+
+![How the algorithm evaluates a doc set]({{site.url}}{{site.baseurl}}/images/hsnw-algorithm.png)
+
+| Variable | Description |
+-- | -- | -- |
+N | Number of documents in the index.
+P | Number of documents in the search set after the filter is applied using the formula: P <= N.
+q | The search vector.
+k | The maximum number of vectors to return in the response.
+
+To learn more about k-NN performance tuning, see [Performance tuning]({{site.url}}{{site.baseurl}}/opensearch/search-plugins/knn/performance-tuning/).
+
+## Filter approaches by use case
+
+Depending on the data set that you are searching, you might choose a different approach to minimize recall or latency. You can create filters that are either:
+* Very restrictive — Returns the least number of documents (For example, 2.5%).
+* Somewhat restrictive — Returns some documents (For example, 38%).
+* Not very restrictive — Returns the most documents (For example, 80%).
+
+The restrictive percentage indicates the amount of documents the filter returns for any given document set in an index.
+
+Number of vectors | Filter restrictive percentage | k | Recall | Latency
+-- | -- | -- | -- | --
+10M | 2.5 | 100 | Scoring script | Scoring script
+10M | 38 | 100 | lucene_filtering | Boolean filter
+10M | 80 | 100 | Scoring script | lucene_filtering
+1M | 2.5 | 100 | lucene_filtering | Scoring script
+1M | 38 | 100 | lucene_filtering | lucene_filtering / Scoring script
+1M | 80 | 100 | Boolean filter | lucene_filtering
+
+In this context, Scoring script is essentially a brute force search, whereas a Boolean filter is an approximate k-NN search with post-filtering.
```

Comment:
Do we need to format the term "Scoring script" so the reader understands it's referring to the latency name in the table? For example, set it in italics or quotations?